### PR TITLE
feat: Introduced Right Subsections for Seamless Topic Access

### PR DESCRIPTION
**Pull request by ankitkumar748846**
#### Changes done:
- [x] Task 1

#### Screenshots/Videos

https://github.com/OneBusAway/onebusaway-docs/assets/109648585/a8a57bbe-a744-4c07-9160-35e65fbccc1d

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Tried squashing the commits into one

### CODE REVIEW

File: frontend/javascript/code_snippets.js [hunk 1]
```diff
@@ -3,6 +3,70 @@ export function enableCodeHighlighting() {
   hljs.highlightAll();
 }
 
+const h1Elements = document.querySelectorAll('h1');
```
Comment (aaronbrethorst) [inline frontend/javascript/code_snippets.js:6]: this code needs to be encapsulated inside of a function that gets invoked from `index.js`

File: frontend/javascript/code_snippets.js [hunk 2]
```diff
@@ -3,6 +3,70 @@ export function enableCodeHighlighting() {
...
+  if(tagName === 'h2') {
+    newItem.classList.add('sidebar-item-h2');
+    newItem.style.color = 'gray';
+    newItem.style.marginLeft = '20px';
+    newItem.style.cursor = 'pointer';
+  }
+  sidebar.appendChild(newItem);
+}
+
+document.addEventListener('DOMContentLoaded', function() {
```
Comment (aaronbrethorst) [inline frontend/javascript/code_snippets.js:31]: only `index.js` should be adding event handlers for `DOMContentLoaded`. Hook it from there in the existing `DOMContentLoaded` callback.

File: frontend/javascript/code_snippets.js [hunk 3]
```diff
@@ -3,6 +3,70 @@ export function enableCodeHighlighting() {
...
+  const sidebarItems = document.querySelectorAll('.sidebar-item-h2');
+
+  sidebarItems.forEach(function(item) {
+     item.addEventListener('mouseenter', function() {
```
Comment (aaronbrethorst) [inline frontend/javascript/code_snippets.js:35]: the color values should be Tailwind classes

File: src/_layouts/default.erb [hunk 4]
```diff
@@ -14,13 +14,13 @@
               <%= render 'site_logo' %>
               <div class="absolute inset-x-0 top-full h-px transition bg-zinc-900/7.5 dark:bg-white/7.5"></div>
             </div>
-
             <%= render 'top_bar' %>
             <%= render 'navbar' %>
           </div>
         </header>
-
+
```
Comment (aaronbrethorst) [inline src/_layouts/default.erb:21]: please delete

File: src/_layouts/default.erb [hunk 5]
```diff
@@ -30,6 +30,9 @@
               </div>
             </article>
           </main>
+          <div class = "hidden sidebar text-xs p-4 2xl:flex flex-col">
```
Comment (aaronbrethorst) [inline src/_layouts/default.erb:33]: I'm not sure what's going on here, but it looks like you have two closing `</div>`s. Please check your indentation too.

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
The feature works great, and I'm excited to get it merged! There is a merge conflict right now, and I have some feedback on the code. Tag me when you've pushed up the changes.

File: src/_layouts/default.erb (hunk 4 above)
Comment (ankitkumar748846) [inline src/_layouts/default.erb:21]:
```suggestion
 
```

**Comment by ankitkumar748846** [comment]
@aaronbrethorst Sir I have implemented the right subsections.  please review it and suggest if any changes required.
Please refer the [PR](https://github.com/OneBusAway/onebusaway-docs/pull/99)

**Pull request closed without merging**